### PR TITLE
TCVP-2553 Fixed statute lookup

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

TCVP-2553
- Fixed bug where the statutes where showing as invalid even though they were valid on the TicketValidation screen
- Per Jira request, removed filtering statutes by ACT - now only by section/subsection/paragraph/sub-paragraph

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts [hunk 1]
```diff
@@ -702,19 +699,17 @@ export class TicketInfoComponent implements OnInit {
     let countForm = this.form.get('violationTicket').get('violationTicketCount' + countNo.toString());
     let parts = fullDescription.split(" "); // act/code fullsection description
 
-    // lookup legal statute from part[1] which should be in legal paragraph form
+    // lookup legal statute from part[0] which should be in legal paragraph form
     if (parts && parts.length > 1) {
-      let foundStatute = this.lookupsService.statutes?.filter(x => x.code === parts[1] && x.actCode === parts[0]).shift();
+      let foundStatute = this.lookupsService.statutes?.find(x => x.code === parts[0]);
```
Comment (KDNeufeld) [inline src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts:704]: .find() is more performant than .filter()
Also removed filtering by parts[0], the ACT.

File: src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts [hunk 2]
```diff
@@ -702,19 +699,17 @@ export class TicketInfoComponent implements OnInit {
...
       if (foundStatute) {
-        countForm.get('actOrRegulationNameCode').setValue(foundStatute.actCode);
         countForm.get('section').setValue(foundStatute.sectionText);
         countForm.get('subsection').setValue(foundStatute.subsectionText);
         countForm.get('paragraph').setValue(foundStatute.paragraphText);
         countForm.get('subparagraph').setValue(foundStatute.subparagraphText);
         countForm.get('description').setValue(foundStatute.shortDescriptionText);
-        countForm.get('fullDescription').setValue(`${foundStatute.actCode} ${foundStatute.code} ${foundStatute.shortDescriptionText}`);
+        countForm.get('fullDescription').setValue(`${foundStatute.code} ${foundStatute.shortDescriptionText}`);
```
Comment (KDNeufeld) [inline src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts:711]: Per Jira, full description now no longer includes the actCode, ie. "MVA".

File: src/frontend/staff-portal/src/app/services/lookups.service.ts [hunk 3]
```diff
@@ -97,7 +97,7 @@ export class LookupsService implements ILookupsService {
   }
 
   public getStatuteString(statute: Statute): string {
-    return `${statute.actCode} ${statute.code} ${statute.descriptionText}`;
+    return `${statute.code} ${statute.descriptionText}`;
```
Comment (KDNeufeld) [inline src/frontend/staff-portal/src/app/services/lookups.service.ts:100]: Full description now no longer starts with actCode, ie. "MVA"

File: src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts (hunk 1 above)
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts:704]: Major difference is filter gives list of items whereas find gives you just 1 record (i.e., first matched record). It depends on the usecase you use.

Comment (KDNeufeld) [reply src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts:704]: Right, but in this case, the code was just using shift(), which pulls the first item off the list. So before it was searching the entire list and then retrieving the first item. Now, it searches up to the first item, returns that and stops.

Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/staff-workbench/ticket-info/ticket-info.component.ts:704]: That's good implementation.

**Review by saikrishnametpalli-nttdata — APPROVED** [review]

**Pull request merged**